layers/+spacemacs/spacemacs-defaults/funcs.el

```diff
@@ -559,30 +559,138 @@ in a split window to the right."
   (split-window-horizontally)
   (other-window 1))
 
-(defun spacemacs/layout-grid ()
-  " Set the layout to a 2x2 grid. "
-  (interactive)
-  (delete-other-windows)
-  (split-window (split-window-right) nil 'below)
-  (split-window-below))
-
-(defun spacemacs/layout-triple-columns ()
-  " Set the layout to triple columns. "
-  (interactive)
-  (delete-other-windows)
-  (dotimes (i 2) (split-window-right))
+
+;; Window Split
+
+(defun spacemacs--window-split-non-ignored-windows ()
+  "Determines the list of windows to be deleted."
+  (seq-filter
+    (lambda (window)
+      (let* ((name (buffer-name (window-buffer window)))
+              (prefixes-matching
+                (seq-filter
+                  (lambda (prefix) (string-prefix-p prefix name))
+                  spacemacs-window-split-ignore-prefixes)))
+        (not prefixes-matching)))
+    (window-list (selected-frame))))
+
+(defun spacemacs/window-split-default-delete ()
+  "Deletes other windows, except a list of excluded ones."
+  (if spacemacs-window-split-ignore-prefixes
+      (let* ((to-delete (spacemacs--window-split-non-ignored-windows)))
+        (when to-delete
+          (select-window (car to-delete))
+          (dolist (window (cdr to-delete)) (delete-window window))))
+    (delete-other-windows)))
+
+(defvar spacemacs-window-split-ignore-prefixes nil
+  "Prefixes for windows that are not deleted when changing split layout.
+
+You can add an entry here by using the following:
+  (add-to-list 'spacemacs-window-split-ignore-prefixes \"Buffer prefix\")")
+
+(defvar spacemacs-window-split-delete-function 'spacemacs/window-split-default-delete
+  "Function used to delete other windows when changing layout.
+
+Used as a callback by the following functions:
+  - spacemacs/window-split-grid
+  - spacemacs/window-split-triple-columns
+  - spacemacs/window-split-double-columns
+  - spacemacs/window-split-single-column
+
+Possible values:
+  - 'spacemacs/window-split-default-delete (default)
+  - 'delete-other-windows
+  - 'treemacs-delete-other-windows (when using the treemacs package)
+  - a lambda: (lambda () (delete-other-windows))
+  - a custom function:
+    (defun my-delete-other-windows () (delete-other-windows))
+    (setq spacemacs-window-split-delete-function 'my-delete-other-windows)")
+
+(defun spacemacs/window-split-grid (&optional purge)
+  "Set the layout to a 2x2 grid.
+
+Uses the funcion defined in `spacemacs-window-split-delete-function' as a means to
+remove windows.
+
+When called with a prefix argument, it uses `delete-other-windows' as a means
+to remove windows, regardless of the value in `spacemacs-window-split-delete-function'."
+  (interactive "P")
+  (let ((ignore-window-parameters t))
```

Comment:
You really want to bind `ignore-window-parameters` only for `delete-other-windows`. Really possible that `spacemacs-window-split-delete-function` wants respect window paramters.
Same goes for the other occurrences.